### PR TITLE
[swift v1]: introduce a TempURLKey argument for objects.CreateTempURLOpts struct

### LINKED ISSUE #2601: [swift v1]: add an ability to specify a custom tempurl key
When creating a tempurl for a foreign account, swift strips the container tempurl keys in response: https://github.com/openstack/swift/blob/629516cbe7eb583c18f429aa04d912c2df24b1a5/swift/proxy/server.py#L303-L315

I propose a new `TempURLKey` member to the `objects.CreateTempURLOpts` struct, which will be used as a default key.

see https://github.com/Lirt/velero-plugin-for-openstack/issues/58

### PR DESCRIPTION
**Pull request by kayrus**
Fixes #2601

### CODE REVIEW

**Comment by kayrus** [comment]
@EmilienM @pierreprinetti looks like something wrong with the CI

**Comment by mandre** [comment]
Master failed with an unrelated error downloading the cirros image during devstack deployment:
```
+ functions:upload_image:142               :   wget --progress=dot:giga -c http://download.cirros-cloud.net/0.5.2/cirros-0.5.2-x86_64-disk.img -O /home/runner/work/gophercloud/gophercloud/devstack/files/cirros-0.5.2-x86_64-disk.img
--2023-03-30 10:39:29--  http://download.cirros-cloud.net/0.5.2/cirros-0.5.2-x86_64-disk.img
Resolving download.cirros-cloud.net (download.cirros-cloud.net)... 64.90.42.85, 2607:f298:6:a036::bd6:a72a
Connecting to download.cirros-cloud.net (download.cirros-cloud.net)|64.90.42.85|:80... connected.
HTTP request sent, awaiting response... 429 Too Many Requests
2023-03-30 10:39:30 ERROR 429: Too Many Requests.
```

However, the other jobs [failed](https://github.com/gophercloud/gophercloud/actions/runs/4563166002/jobs/8051322961?pr=2602#step:5:1051) for a legitimate reason:
```
 === RUN   TestObjects
    objects_test.go:39: Create object headers: &{ContentLength:0 ContentType:text/html; charset=UTF-8 Date:2023-03-30 10:46:27 +0000 GMT TransID:tx505c20822268478db237c-0064256883}
    convenience.go:36: Failure in objects_test.go, line 96: unexpected error "Unable to obtain the Temp URL key."
--- FAIL: TestObjects (0.53s)
```

**Comment by kayrus** [comment]
@mandre funny, it looks like tempurl acceptance tests never worked.

**Comment by kayrus** [comment]
@mandre ready to review.

**Comment by kayrus** [comment]
@EmilienM @pierreprinetti @mandre kindly ping.

File: acceptance/openstack/objectstorage/v1/objects_test.go [hunk 1]
```diff
@@ -95,12 +99,37 @@ func TestObjects(t *testing.T) {
 		})
 		th.AssertNoErr(t, err)
 
-		resp, err := http.Get(objURLs[i])
+		resp, err := client.ProviderClient.HTTPClient.Get(objURLs[i])
```
Comment (mandre) [inline acceptance/openstack/objectstorage/v1/objects_test.go:102]: Shouldn't we keep using an unauthenticated client for downloading the objects using the temp URL?

Comment (kayrus) [reply acceptance/openstack/objectstorage/v1/objects_test.go:102]: it's already unauthenticated. if you take a look on the debug code you can see that there are no headers used. this is just more convenient way to reuse the existing http client on the low level + get a bonus for http request/response debug.

**Review by mandre — APPROVED** [review]
LGTM

**Pull request merged**